### PR TITLE
chore(iam): improve iam user console access check

### PR DESCRIPTION
**Pull request by puchy22**
### Context

Fixes  #3176

Ensuer if user currently has access to console before checking the days of last log

### Description

Add new property to the modelo of User in IAM client, change checks logic comprobation and change test case logic adding a new password to users created or in last test change the status_extended.

### License

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: prowler/providers/aws/services/iam/iam_service.py [hunk 1]
```diff
@@ -274,16 +274,21 @@ def __get_users__(self):
...
-                                    arn=user["Arn"],
-                                    password_last_used=user["PasswordLastUsed"],
-                                )
+                        try:
+                            user_login_profile = self.client.get_login_profile(
+                                UserName=user["UserName"]
                             )
+                        except self.client.exceptions.NoSuchEntityException:
+                            user_login_profile = None
+
```
Comment (jfagoagas) [inline prowler/providers/aws/services/iam/iam_service.py:288]: Please add `except Exception as error:` here to catch any other exception with a log level ERROR and setting the `user_login_profile` to `None`.

File: prowler/providers/aws/services/iam/iam_user_console_access_unused/iam_user_console_access_unused.py [hunk 2]
```diff
@@ -31,10 +31,7 @@ def execute(self) -> Check_Report_AWS:
                     report.status_extended = f"User {user.name} has logged in to the console in the past {maximum_expiration_days} days ({time_since_insertion.days} days)."
             else:
                 report.status = "PASS"
-                report.status_extended = (
-                    f"User {user.name} does not have a console password or is unused."
-                )
+                report.status_extended = f"User {user.name} does not have a console access enabled or is unused."
```
Comment (jfagoagas) [inline prowler/providers/aws/services/iam/iam_user_console_access_unused/iam_user_console_access_unused.py]:
```suggestion
                report.status_extended = f"User {user.name} does not have console access enabled or is unused."
```

File: tests/providers/aws/services/iam/iam_user_console_access_unused/iam_user_console_access_unused_test.py [hunk 3]
```diff
@@ -123,7 +126,7 @@ def test_iam_user_not_logged(self):
                 assert result[0].status == "PASS"
                 assert (
                     result[0].status_extended
-                    == f"User {user} does not have a console password or is unused."
+                    == f"User {user} does not have a console access enabled or is unused."
```
Comment (jfagoagas) [inline tests/providers/aws/services/iam/iam_user_console_access_unused/iam_user_console_access_unused_test.py]:
```suggestion
                    == f"User {user} does not have console access enabled or is unused."
```

**Review by jfagoagas — CHANGES_REQUESTED** [review]
Great improvement @puchy22 👏  Please review my comments when you get a chance. Thanks!

File: prowler/providers/aws/services/iam/iam_service.py (hunk 1 above)
Comment (jfagoagas) [inline prowler/providers/aws/services/iam/iam_service.py:283]: You need to check this way the exceptions in AWS, the `self.client.exceptions.NoSuchEntityException` does not work and right now I'm not sure why.
```suggestion
                        except ClientError as error:
                            user_login_profile = None
                            if error.response["Error"]["Code"] == "NoSuchEntity":
                                logger.warning(
                                    f"{self.region} -- {error.__class__.__name__}[{error.__traceback__.tb_lineno}]: {error}"
                                )
                            else:
                                logger.error(
                                    f"{self.region} -- {error.__class__.__name__}[{error.__traceback__.tb_lineno}]: {error}"
                                )

                        except Exception as error:
                            user_login_profile = None
                            logger.error(
                                f"{self.region} -- {error.__class__.__name__}[{error.__traceback__.tb_lineno}]: {error}"
                            )

```

Comment (puchy22) [reply prowler/providers/aws/services/iam/iam_service.py:288]: Added 🫡

**Review by jfagoagas — APPROVED** [review]
Awesome!! Thanks for taking care of this 🚀

**Pull request merged**